--- avahi-0.6.32/avahi-core/server.c	2016-11-19 19:48:33.295906822 +0100
+++ avahi-0.6.32.b/avahi-core/server.c	2016-11-19 20:00:12.937914509 +0100
@@ -976,7 +976,9 @@
             avahi_dns_packet_get_field(p, AVAHI_DNS_FIELD_ANCOUNT) == 0 ||
             avahi_dns_packet_get_field(p, AVAHI_DNS_FIELD_NSCOUNT) != 0) {
 
+            /* Avoid flooding
             avahi_log_debug("Invalid response packet from host %s.", avahi_address_snprint(t, sizeof(t), src_address));
+            */
             return;
         }
 
