Index: metacity-2.34.13/src/core/display-private.h
===================================================================
--- metacity-2.34.13.orig/src/core/display-private.h
+++ metacity-2.34.13/src/core/display-private.h
@@ -362,6 +362,7 @@ MetaDisplay* meta_display_for_x_display
 MetaDisplay* meta_get_display            (void);
 
 Cursor         meta_display_create_x_cursor (MetaDisplay *display,
+                                             Window       window,
                                              MetaCursor   cursor);
 
 void     meta_display_set_grab_op_cursor (MetaDisplay *display,
Index: metacity-2.34.13/src/core/display.c
===================================================================
--- metacity-2.34.13.orig/src/core/display.c
+++ metacity-2.34.13/src/core/display.c
@@ -3029,8 +3029,20 @@ meta_display_xwindow_is_a_no_focus_windo
   return is_a_no_focus_window;
 }
 
+static Cursor
+create_blank_cursor (MetaDisplay *display, Window window)
+{
+  Cursor xcursor;
+  XColor dummy;
+  static char data[1] = {0};
+  Pixmap blank = XCreateBitmapFromData(display->xdisplay, window, data, 1, 1);
+  xcursor = XCreatePixmapCursor(display->xdisplay, blank, blank, &dummy, &dummy, 0, 0);
+  return xcursor;
+}
+
 Cursor
 meta_display_create_x_cursor (MetaDisplay *display,
+                              Window window,
                               MetaCursor cursor)
 {
   Cursor xcursor;
@@ -3041,41 +3053,19 @@ meta_display_create_x_cursor (MetaDispla
     case META_CURSOR_DEFAULT:
       glyph = XC_left_ptr;
       break;
+    case META_CURSOR_BLANK:
     case META_CURSOR_NORTH_RESIZE:
-      glyph = XC_top_side;
-      break;
     case META_CURSOR_SOUTH_RESIZE:
-      glyph = XC_bottom_side;
-      break;
     case META_CURSOR_WEST_RESIZE:
-      glyph = XC_left_side;
-      break;
     case META_CURSOR_EAST_RESIZE:
-      glyph = XC_right_side;
-      break;
     case META_CURSOR_SE_RESIZE:
-      glyph = XC_bottom_right_corner;
-      break;
     case META_CURSOR_SW_RESIZE:
-      glyph = XC_bottom_left_corner;
-      break;
     case META_CURSOR_NE_RESIZE:
-      glyph = XC_top_right_corner;
-      break;
     case META_CURSOR_NW_RESIZE:
-      glyph = XC_top_left_corner;
-      break;
     case META_CURSOR_MOVE_OR_RESIZE_WINDOW:
-      glyph = XC_fleur;
-      break;
     case META_CURSOR_BUSY:
-      glyph = XC_watch;
-      break;
-      
     default:
-      g_assert_not_reached ();
-      glyph = 0; /* silence compiler */
-      break;
+      return create_blank_cursor(display, window);
     }
   
   xcursor = XCreateFontCursor (display->xdisplay, glyph);
@@ -3085,6 +3075,7 @@ meta_display_create_x_cursor (MetaDispla
 
 static Cursor
 xcursor_for_op (MetaDisplay *display,
+                Window       window,
                 MetaGrabOp   op)
 {
   MetaCursor cursor = META_CURSOR_DEFAULT;
@@ -3135,7 +3126,7 @@ xcursor_for_op (MetaDisplay *display,
 
   if (cursor == META_CURSOR_DEFAULT)
     return None;
-  return meta_display_create_x_cursor (display, cursor);
+  return meta_display_create_x_cursor (display, window, cursor);
 }
 
 void
@@ -3148,7 +3139,7 @@ meta_display_set_grab_op_cursor (MetaDis
 {
   Cursor cursor;
 
-  cursor = xcursor_for_op (display, op);
+  cursor = xcursor_for_op (display, screen->xroot, op);
 
 #define GRAB_MASK (PointerMotionMask |                          \
                    ButtonPressMask | ButtonReleaseMask |        \
Index: metacity-2.34.13/src/core/frame.c
===================================================================
--- metacity-2.34.13.orig/src/core/frame.c
+++ metacity-2.34.13/src/core/frame.c
@@ -407,7 +407,7 @@ meta_frame_set_screen_cursor (MetaFrame
     XUndefineCursor (frame->window->display->xdisplay, frame->xwindow);
   else
     { 
-      xcursor = meta_display_create_x_cursor (frame->window->display, cursor);
+      xcursor = meta_display_create_x_cursor (frame->window->display, frame->xwindow, cursor);
       XDefineCursor (frame->window->display->xdisplay, frame->xwindow, xcursor);
       XFlush (frame->window->display->xdisplay);
       XFreeCursor (frame->window->display->xdisplay, xcursor);
Index: metacity-2.34.13/src/core/screen.c
===================================================================
--- metacity-2.34.13.orig/src/core/screen.c
+++ metacity-2.34.13/src/core/screen.c
@@ -533,7 +533,7 @@ meta_screen_new (MetaDisplay *display,
   
   reload_xinerama_infos (screen);
   
-  meta_screen_set_cursor (screen, META_CURSOR_DEFAULT);
+  meta_screen_set_cursor (screen, META_CURSOR_BLANK);
 
   /* Handle creating a no_focus_window for this screen */  
   screen->no_focus_window =
@@ -1179,7 +1179,7 @@ meta_screen_set_cursor (MetaScreen *scre
 
   screen->current_cursor = cursor;
   
-  xcursor = meta_display_create_x_cursor (screen->display, cursor);
+  xcursor = meta_display_create_x_cursor (screen->display, screen->xroot, cursor);
   XDefineCursor (screen->display->xdisplay, screen->xroot, xcursor);
   XFlush (screen->display->xdisplay);
   XFreeCursor (screen->display->xdisplay, xcursor);
@@ -1190,7 +1190,7 @@ meta_screen_update_cursor (MetaScreen *s
 {
   Cursor xcursor;
 
-  xcursor = meta_display_create_x_cursor (screen->display, 
+  xcursor = meta_display_create_x_cursor (screen->display, screen->xroot,
 					  screen->current_cursor);
   XDefineCursor (screen->display->xdisplay, screen->xroot, xcursor);
   XFlush (screen->display->xdisplay);
Index: metacity-2.34.13/src/include/common.h
===================================================================
--- metacity-2.34.13.orig/src/include/common.h
+++ metacity-2.34.13/src/include/common.h
@@ -147,6 +147,7 @@ typedef enum
 
 typedef enum
 {
+  META_CURSOR_BLANK,
   META_CURSOR_DEFAULT,
   META_CURSOR_NORTH_RESIZE,
   META_CURSOR_SOUTH_RESIZE,
