diff -uNr e2fsprogs-1.42.8/e2fsck/problem.c e2fsprogs-1.42.8-2/e2fsck/problem.c
--- e2fsprogs-1.42.8/e2fsck/problem.c	2013-06-21 04:43:42.000000000 +0200
+++ e2fsprogs-1.42.8-2/e2fsck/problem.c	2017-08-10 12:16:58.396748166 +0200
@@ -1819,7 +1819,7 @@
 	ext2_filsys fs = ctx->fs;
 	struct e2fsck_problem *ptr;
 	struct latch_descr *ldesc = 0;
-	const char *message;
+	char message[512];
 	int		def_yn, answer, ans;
 	int		print_answer = 0;
 	int		suppress = 0;
@@ -1901,9 +1901,10 @@
 			fflush(stdout);
 		}
 	}
-	message = ptr->e2p_description;
-	if (*message)
-		message = _(message);
+        char error_code[27];
+        sprintf(error_code, "( Problem Code 0x%06x ) ", code);
+        strncpy(message, error_code, sizeof(message));
+        strncpy(message + sizeof(error_code) -1, ptr->e2p_description, sizeof(message) - sizeof(error_code));
 	if (!suppress) {
 		if ((ctx->options & E2F_OPT_PREEN) &&
 		    !(ptr->flags & PR_PREEN_NOHDR)) {
