--- a/os/WaitFor.c	2012-06-28 19:06:18.000000000 +0200
+++ b/os/WaitFor.c	2017-05-29 17:55:08.911946249 +0200
@@ -140,7 +140,7 @@
  *     For more info on ClientsWithInput, see ReadRequestFromClient().
  *     pClientsReady is an array to store ready client->index values into.
  *****************/
-
+static int first_run = 0;
 int
 WaitForSomething(int *pClientsReady)
 {
@@ -206,10 +206,13 @@
             }
             XFD_COPYSET(&AllSockets, &LastSelectMask);
         }
-
-        BlockHandler((pointer) &wt, (pointer) &LastSelectMask);
-        if (NewOutputPending)
-            FlushAllOutput();
+        if(first_run > 1){
+            first_run++;
+        } else {
+            BlockHandler((pointer) &wt, (pointer) &LastSelectMask);
+            if (NewOutputPending)
+                FlushAllOutput();
+        }
         /* keep this check close to select() call to minimize race */
         if (dispatchException)
             i = -1;
