diff -uNr xorg-server-1.19.3-2/os/WaitFor.c xorg-server-1.19.3/os/WaitFor.c
--- xorg-server-1.19.3-2/os/WaitFor.c	2018-02-20 12:50:16.779995672 +0100
+++ xorg-server-1.19.3/os/WaitFor.c	2018-02-20 12:51:51.265099360 +0100
@@ -180,6 +180,8 @@
  *     pClientsReady is an array to store ready client->index values into.
  *****************/
 
+static int first_run = 0;
+
 Bool
 WaitForSomething(Bool are_ready)
 {
@@ -216,9 +218,14 @@
         else
             timeout = check_timers();
 
-        BlockHandler(&timeout);
-        if (NewOutputPending)
-            FlushAllOutput();
+        if(first_run > 1){
+            first_run++;
+        } else {
+            BlockHandler(&timeout);
+            if (NewOutputPending)
+                FlushAllOutput();
+        }
+
         /* keep this check close to select() call to minimize race */
         if (dispatchException)
             i = -1;
