diff -uNr chromium-65.0.3325.162-2/ui/base/ime/input_method_auralinux.cc chromium-65.0.3325.162/ui/base/ime/input_method_auralinux.cc
--- chromium-65.0.3325.162-2/ui/base/ime/input_method_auralinux.cc	2018-07-09 17:09:56.478393777 +0200
+++ chromium-65.0.3325.162/ui/base/ime/input_method_auralinux.cc	2018-07-09 17:11:55.493962713 +0200
@@ -232,11 +232,11 @@
   text_input_type_ = GetTextInputType();
 
   // We only focus in |context_| when the focus is in a textfield.
-  if (old_text_input_type != TEXT_INPUT_TYPE_NONE &&
-      text_input_type_ == TEXT_INPUT_TYPE_NONE) {
+  if (text_input_type_ == TEXT_INPUT_TYPE_NONE && old_text_input_type != TEXT_INPUT_TYPE_NONE )
+    context_->Blur();
+
+  if (text_input_type_ != TEXT_INPUT_TYPE_NONE) {
     context_->Blur();
-  } else if (old_text_input_type == TEXT_INPUT_TYPE_NONE &&
-             text_input_type_ != TEXT_INPUT_TYPE_NONE) {
     context_->Focus();
   }
 
@@ -264,6 +264,10 @@
   }
 }
 
+void InputMethodAuraLinux::ShowImeIfNeeded() {
+  UpdateContextFocusState();
+}
+
 void InputMethodAuraLinux::OnTextInputTypeChanged(
     const TextInputClient* client) {
   UpdateContextFocusState();
diff -uNr chromium-65.0.3325.162-2/ui/base/ime/input_method_auralinux.h chromium-65.0.3325.162/ui/base/ime/input_method_auralinux.h
--- chromium-65.0.3325.162-2/ui/base/ime/input_method_auralinux.h	2018-07-09 17:09:56.482393762 +0200
+++ chromium-65.0.3325.162/ui/base/ime/input_method_auralinux.h	2018-07-09 17:09:59.878381487 +0200
@@ -31,6 +31,7 @@
                                 NativeEventResult* result) override;
   ui::EventDispatchDetails DispatchKeyEvent(ui::KeyEvent* event) override;
   void OnTextInputTypeChanged(const TextInputClient* client) override;
+  void ShowImeIfNeeded() override;
   void OnCaretBoundsChanged(const TextInputClient* client) override;
   void CancelComposition(const TextInputClient* client) override;
   bool IsCandidatePopupOpen() const override;
